AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -2583,12 +2664,18 @@ - (void)setCalculatedDisplayNodeLayout:(std::shared_ptr<ASDisplayNodeLayout>)dis
 - (CGSize)calculatedSize
 {
   ASDN::MutexLocker l(__instanceLock__);
+  if (_pendingDisplayNodeLayout != nullptr) {
+    return _pendingDisplayNodeLayout->layout.size;
+  }
   return _calculatedDisplayNodeLayout->layout.size;
 }
 
 - (ASSizeRange)constrainedSizeForCalculatedLayout
 {
   ASDN::MutexLocker l(__instanceLock__);
+  if (_pendingDisplayNodeLayout != nullptr) {
+    return _pendingDisplayNodeLayout->constrainedSize;
+  }
   return _calculatedDisplayNodeLayout->constrainedSize;
```

Comment:
By default it will return `{{0, 0}, {0, 0}}`, that's passed in in the initializer of the `ASDisplayNodeLayout`:

```objc
  ASDisplayNodeLayout()
  : layout(nil), constrainedSize({{0, 0}, {0, 0}}), parentSize({0, 0}), _dirty(YES) {};
```